src/components/LinkableText.tsx

```diff
@@ -1,14 +1,21 @@
 import React from 'react';
-import {Linking, Text} from 'react-native';
+import {Linking, StyleProp, Text, TextStyle} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
 
+import TopicDetectionOnChat from './Text/TopicDetectionOnChat';
 import {COLORS} from '../utils/theme';
 
 interface Props {
   text: string;
-  style?: object;
+  style?: StyleProp<TextStyle>;
+  multiline?: false;
+  withTopicDetection?: boolean;
+  isFirstLine?: boolean;
 }
 
 export const LinkableText: React.FC<Props> = ({text, ...props}) => {
+  const navigation = useNavigation();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useNavigation` hook is used but there's no check to ensure that the component is within a navigation context. If it's not, this could lead to an error. Consider adding error handling for this case.
```diff
  const navigation = useNavigation();
+ if (!navigation) {
+   console.error('LinkableText must be used within a navigation context');
+   return null;
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->